File: app/src/test/java/com/mysticwind/linenotificationsupport/chatname/model/ChatTest.java
package com.mysticwind.linenotificationsupport.chatname.model;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;

import org.junit.Test;

public class ChatTest {

    @Test
    public void equalityUsesIdAndName() {
        Chat left = new Chat("id-1", "name");
        Chat right = new Chat("id-1", "name");
        Chat different = new Chat("id-2", "name");

        assertEquals(left, right);
        assertEquals(left.hashCode(), right.hashCode());
        assertNotEquals(left, different);
    }
}

File: app/src/test/java/com/mysticwind/linenotificationsupport/chatname/model/ChatTest.kt
package com.mysticwind.linenotificationsupport.chatname.model

import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotEquals
import org.junit.Test

class ChatTest {

    @Test
    fun equalityUsesIdAndName() {
        val left = Chat("id-1", "name")
        val right = Chat("id-1", "name")
        val different = Chat("id-2", "name")

        assertEquals(left, right)
        assertEquals(left.hashCode(), right.hashCode())
        assertNotEquals(left, different)
    }
}
